src/wallet/rpc/wallet.cpp

```diff
@@ -344,6 +344,11 @@ static RPCHelpMan createwallet()
     uint64_t flags = 0;
     if (!request.params[1].isNull() && request.params[1].get_bool()) {
         flags |= WALLET_FLAG_DISABLE_PRIVATE_KEYS;
+        // We want to set BLANK when DISABLE_PRIVATE_KEYS, but only if the user did not explicitly do blank=false
+        // It can be set when blank is not provided; when it is provided, we handle it later
```

Comment:
shouldn't the created wallet be blank whenever `DISABLE_PRIVATE_KEYS` is set?

Reply:
As noted in the OP, `blank=false` overrides the default behavior.

Reply:
Yes but does it make sense to create a wallet that is "not empty" and without private keys? I may be missing something

Reply:
> However, for backwards compatibility, users can choose to override this in the RPC by doing blank=False.

I guess that's the reason.

Reply:
The blank flag is basically a no-op for wallets without private keys. It's not necessarily a useful distinction.

Reply:
I agree with @achow101 . Initially I was concerned that a bug in a future change might be introduced due to the apparent contradiction between disable_private_keys=true and blank=false, but after reading the tests, I understand that this might be posible after importing ADDRESSes and/or other elements. Am I right about this conclusion? Or am I missing something?

Reply:
Yes, `blank` will be unset once anything is imported or generated. The purpose of `blank` is to prevent the wallet from automatically generating keys for blank wallets since they appear the same as newly created wallets to the creation/setup code that runs during wallet loading. Since disabling private keys means that nothing can be generated anyways, blank isn't actually doing anything for such wallets. Additionally, once anything is imported, blank will be unset as well.